src/main/java/com/mazawrath/beanbot/commands/MinesweeperCommand.java

```diff
@@ -0,0 +1,130 @@
+package com.mazawrath.beanbot.commands;
+
+import com.mazawrath.beanbot.utilities.Points;
+import com.mazawrath.beanbot.utilities.SentryLog;
+import de.btobastian.sdcf4j.Command;
+import de.btobastian.sdcf4j.CommandExecutor;
+import io.sentry.Sentry;
+import org.apache.commons.lang3.StringUtils;
+import org.javacord.api.DiscordApi;
+import org.javacord.api.entity.channel.ServerTextChannel;
+import org.javacord.api.entity.message.Message;
+import org.javacord.api.entity.server.Server;
+import org.javacord.api.entity.user.User;
+
+import java.util.Arrays;
+import java.util.Random;
+
+public class MinesweeperCommand implements CommandExecutor {
+    private Points points;
+
+    public MinesweeperCommand(Points points) {
+        this.points = points;
+    }
+
+    @Command(
+            aliases = {"minesweeper"},
+            usage = "minesweeper [size] [mines]",
+            description = "Creates a minesweeper field with custom size options.",
+            privateMessages = false
+    )
+
+    public void onCommand(String[] args, DiscordApi api, ServerTextChannel serverTextChannel, User author, Server server, Message message) {
+        SentryLog.addContext(args, author, server);
+
+        int size;
+        int mines;
+
+        if (args.length >= 2) {
+            if (StringUtils.isNumeric(args[0]) && StringUtils.isNumeric(args[1])) {
+                size = Integer.parseInt(args[0]);
+                mines = Integer.parseInt(args[1]);
+            } else {
+                serverTextChannel.sendMessage("Invalid numbers.");
+                return;
+            }
+        } else {
+            size = 10;
+            mines = size * size / 6;
+        }
+
+        if (points.removePoints(author.getIdAsString(), api.getYourself().getIdAsString(), server.getIdAsString(), Points.COMMAND_COST)) {
+            serverTextChannel.sendMessage(new MinesweeperField(ensureInRange(size, 1,14), ensureInRange(mines, 1, size*size)).toDiscordString());
```

Comment:
if I send
minesweeper size=200 mines=195
ensureInRange(200, 1,14) => (200 < 1) ? 1: ((200 > 14) ? 14 : 200) => 14
ensureInRange(195, 1, 40,000) =>(195 < 1) ? 1 : ((195 > 40,000) ? 40,000: 195) => 195

new MinesweeperField(14, 195)
[ (14 * 14 = 196) < 195 = False ] => continues

=> Minesweeper with 195 mines out of 196 fields